contracts/Vault2.sol

```diff
@@ -1,6 +1,32 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+import "./Vault.sol";
+import "hardhat/console.sol";
+
+
 contract Vault2 {
-    
+
+  Vault token;
+
+     constructor (address _tokenAddress) {
+         token=Vault(_tokenAddress);
+     }
+
+    function mint(uint256 amount) external payable {
+        uint ethAmount= (uint)(msg.value/10**18);      
```

Comment:
Best to keep this in wei itself. In case user sends 0.99 ether, this amount here will round off to 0